src/smt.rs

```diff
@@ -1114,59 +1136,47 @@ impl SparseMerkleTree {
             }
             bin_offset += new_offset
         }
-        let query_height = h as usize;
 
-        let mut ancestor_hashes = VecDeque::new();
-        let mut sibling_hashes = VecDeque::new();
-        let mut binary_bitmap: Vec<bool> = vec![];
-        let current_node = current_node.unwrap();
-
-        let max_structure = current_subtree
-            .structure
-            .iter()
-            .max()
-            .ok_or_else(|| SMTError::Unknown(String::from("Invalid structure")))?;
-
-        let extra = QueryHashesExtraInfo::new(
-            *max_structure,
-            current_node.index,
-            current_subtree.nodes.len(),
-        );
-        let info = QueryHashesInfo::new(
-            &current_subtree.nodes,
-            &current_subtree.structure,
-            &mut sibling_hashes,
-            &mut ancestor_hashes,
-            &mut binary_bitmap,
-            extra,
-        );
-        calculate_query_hashes(info);
+        Ok((current_node.unwrap(), h as usize))
+    }
 
-        if current_node.kind == NodeKind::Empty {
+    fn generate_result(
```

Comment:
maybe the name is too generic? same for below `calc_extra_info`

Reply:
rename the following functions:

- calc_extra_info -> make_query_hashes_extra_info

- generate_result -> returns_query_proof

Reply:
i think `returns` seems bit strange maybe because `return` is reserved keyword.
how about `calc_query_hashes_extra_info` and `calc_query_proof_from_result`

Reply:
I have same problem with those name, new suggestions are better.